### PR TITLE
Add pagination to all pickers #1632

### PR DESCRIPTION
**Pull request by bastianallgeier**
## Describe the PR

Pagination is now implemented for all pickers (pages, users, files) and all pickers have new underlying PHP classes to refactor their spaghetti code from the fields definitions.

## Related issues

- #1632

## Ready?

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Fixed code style issues with CS fixer and `composer fix`
- [x] Added in-code documentation (if neeed)

### CODE REVIEW

File: src/Cms/FilePicker.php [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        } elseif (is_a($model, 'Kirby\Cms\File') === true) {
+            $query = 'file.siblings';
+        } else {
+            $query = $model::CLASS_ALIAS . '.files';
+        }
+
+        // fetch all files for the picker
+        $files = $model->query($query);
+
+        // help mitigate some typical query usage issues
```
Comment (distantnative) [inline src/Cms/FilePicker.php:89]: Does this make sense in the context of the FilePicker? Shouldn't it convert to files not children?

File: src/Cms/FilePicker.php [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+    /**
+     * Converts all given files to an associative
+     * array that is already optimized for the
+     * panel picker component.
+     *
+     * @param \Kirby\Cms\Files|null $files
+     * @return array
+     */
+    public function filesToArray($files): array
```
Comment (distantnative) [inline src/Cms/FilePicker.php]: `$files=null`?

File: src/Cms/UserPicker.php [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+     *
+     * @param array $params
+     */
+    public function __construct(array $params = [])
+    {
+        // default params
+        $defaults = [
+            // image settings (ratio, cover, etc.)
+            'image' => [],
+            // query template for the page info field
```
Comment (distantnative) [inline src/Cms/UserPicker.php]: `user info field`

File: src/Cms/UserPicker.php [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+            'info' => false,
+            // number of pages displayed per pagination page
```
Comment (distantnative) [inline src/Cms/UserPicker.php]: `number of user`

**Comment by distantnative** [comment]
Added some small fixes and one review comment - other than that great!

File: src/Cms/FilePicker.php [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        // pages by returning their children
+        if (is_a($files, 'Kirby\Cms\Site') === true) {
+            $files = $files->children();
+        } elseif (is_a($files, 'Kirby\Cms\Page') === true) {
+            $files = $files->children();
+        } elseif (is_a($files, 'Kirby\Cms\User') === true) {
+            $files = $files->children();
+        } elseif (is_a($files, 'Kirby\Cms\Files') === false) {
+            throw new InvalidArgumentException('Your query must return a set of files');
+        }
```
Comment (afbora) [inline src/Cms/FilePicker.php]: Repeated `$files = $files->children();` 🤔 

I think there's a mistake here. If the query is one of these, the picker will list the pages.

Comment (distantnative) [reply src/Cms/FilePicker.php]: @afbora I think we both found the same bug in this :)

Comment (afbora) [reply src/Cms/FilePicker.php]: You right! When I focused on the bug, i missed your review 👍

**Comment by distantnative** [comment]
But I am confused why our PHPUnit test is failing because of the `escape` helper.

**Comment by bastianallgeier** [comment]
I think the PHPunit escape issue comes from another commit.

**Comment by distantnative** [comment]
@bastianallgeier yes Lukas mentioned that. So only the section @afbora and I marked as weird

File: src/Cms/FilePicker.php (hunk 5 above)
Comment (bastianallgeier) [reply src/Cms/FilePicker.php]: Ok, that was stupid! It took me a while to find what you mean. I was totally blind :)

**Pull request merged**